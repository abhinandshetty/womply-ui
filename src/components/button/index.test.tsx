import React from 'react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'
import { Button } from '.'

describe('Button component test suite', () => {
  test('Button renders as expected', () => {
    const tree = renderer.create(<Button label="Custom Button" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('check if button label is rendered correctly', () => {
    const { getByText } = render(<Button label="Custom Button" />)

    expect(getByText('Custom Button')).toBeInTheDocument()
  })

  test("check if 'onClick' function is called", () => {
    const onClick = jest.fn()
    const { getByTestId } = render(<Button label="Custom Button" onClick={onClick} />)

    const button = getByTestId('btn')
    fireEvent.click(button)
    expect(onClick).toBeCalled()
  })
})
