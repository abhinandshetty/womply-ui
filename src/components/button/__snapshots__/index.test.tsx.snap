// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test suite Button renders as expected 1`] = `
<button
  data-testid="btn"
>
  Custom Button
</button>
`;
